File: js/delete-digit.js
function deleteDigit(n){
    let s = String(n).split('').map(Number)
    return Math.max(...s.map((el, i) => {
        let r = s.slice(); 
        r.splice(i, 1);
        return Number(r.join(''))
        }))
}

let n = 152

console.log(`For the integer ${n} the maximum maximal number you can obtain by deleting exactly one digit from it is:`, deleteDigit(n))

File: go/delete-digit/delete-digit.go
package main

import (
	"fmt"
	"strconv"
)

func deleteDigit(n int) int {
	s := strconv.Itoa(n)
	maxNum := 0

	for i := 0; i < len(s); i++ {
		r := s[:i] + s[i+1:]
		num, _ := strconv.Atoi(r)
		if num > maxNum {
			maxNum = num
		}
	}


	return maxNum
}

func main() {
	n := 152

	fmt.Printf("For the integer %d the maximum maximal number you can obtain by deleting exactly one digit from it is: %d", n, deleteDigit(n))
}